packages/next/client/index.tsx

```diff
@@ -684,17 +685,19 @@ if (process.env.__NEXT_RSC) {
     }
 
     const root = response.readRoot()
-    return root
+    return React.cloneElement(root, restProps)
```

Comment:
I doubt if this will actually work though, the RSC root is rendered on the server side and you can't re-render it with new props on the client side. The components are not even bundled here.